invenio_rdm_records/records/custom_fields.py

```diff
@@ -14,33 +14,115 @@
 from flask import current_app
 from werkzeug.local import LocalProxy
 
-from invenio_records.systemfields.relations import RelationBase
+from invenio_records.systemfields.relations import (
+    RelationsField,
+    RelationBase,
+    RelationsMapping,
+)
 from invenio_vocabularies.records.api import Vocabulary
 
 
-class CustomFieldsRelation(RelationBase):
+class CustomFieldsRelation(RelationsField):
     """Relation field to manage custom fields."""
 
     def __init__(self):
         """Initialize the field."""
         super().__init__()
-        self._fields = LocalProxy(
-            lambda: self._load_custom_fields_relations()
-        )
+        self._fields = LocalProxy(lambda: self._load_custom_fields_relations())
 
     def _load_custom_fields_relations(self):
-
         cfs = current_app.config.get("RDM_RECORDS_CUSTOM_FIELDS", {})
 
         relations = {}
         for cf in cfs.values():
             # FIXME: how to abstract this so relation_cls is not present in BaseCF
             if cf.relation_cls:
                 relations[cf.name] = cf.relation_cls(
-                f"custom.{cf.name}",
-                keys=["title", "props", "icon"],
-                pid_field=Vocabulary.pid.with_type_ctx(cf.vocabulary_id),
-                cache_key=cf.vocabulary_id,
-            )
+                    f"custom.{cf.name}",
+                    keys=["title", "props", "icon"],
+                    pid_field=Vocabulary.pid.with_type_ctx(cf.vocabulary_id),
+                    cache_key=cf.vocabulary_id,
+                )
 
         return relations
+
+    def __set__(self, instance, values):
+        """Setting the attribute."""
+        raise ValueError("Cannot set this field directly.")
+
+
+class MultiRelationsField(RelationsField):
+    """Relations field for connections to external entities."""
+
+    def __init__(self, **fields):
+        """Initialize the field."""
+        assert all(
+            isinstance(f, RelationBase) or isinstance(f, RelationsField)
+            for f in fields.values()
+        )
+        self._fields = {
+            key: field
+            for (key, field) in fields.items()
+            if isinstance(field, RelationBase)
+        }
+        self._relation_fields = {
+            key: field
```

Comment:
are we using the `key` anywhere? or could it be just a list? (curiosity since it does not change any other implementeation)